docs/specifications/broker-authentication.md

```diff
@@ -0,0 +1,107 @@
+# Broker Authentication
+
+This document attempts to outline the high-level plan for adding authentication
+to the broker api routes.
+
+When talking to service brokers, the [Service Catalog uses basic auth](#service-catalog-basic-auth). The Ansible Service Broker will implement Basic Auth using a secret shared between the catalog and the broker. The authentication will be configurable allowing a user i.e. developer to disable it.
+
+The plan for this sprint is to do the following:
+
+* make auth routes configurable by enhancing the configuration
+```
+auth:
+   - type: basic | INSERT OTHER TYPE HERE
+   - enable: true | false
+```
+  * update the following files to reflect the new entry:
+      * [deploy-ansible-service-broker.template.yaml](https://github.com/openshift/ansible-service-broker/blob/master/templates/deploy-ansible-service-broker.template.yaml)
+      * [prep_local_devel_env.sh](https://github.com/openshift/ansible-service-broker/blob/master/scripts/prep_local_devel_env.sh) script
+      * [example-config.yaml](https://github.com/openshift/ansible-service-broker/blob/master/etc/example-config.yaml)
+      * [README.md](https://github.com/openshift/ansible-service-broker/blob/master/README.md)
+      * docs directory:
+          * [deployment.md](https://github.com/openshift/ansible-service-broker/blob/master/docs/deployment.md)
+          * [config.md](https://github.com/openshift/ansible-service-broker/blob/master/docs/config.md)
+          * [local_deployment.md](https://github.com/openshift/ansible-service-broker/blob/master/docs/local_deployment.md)l
+      * [`make deploy`](https://github.com/openshift/ansible-service-broker/blob/master/Makefile#L57-L58) does not need updating because it uses the above template.
+  * [catasb](https://github.com/fusor/catasb/tree/dev)
+      * pulls down the `deploy-ansible-service-broker.template.yaml`
+      * may have to update all.yml in case we need to override the auth setting
+
+* create a secret that both the broker and service catalog can use
+
+* secret rotation is desired but needs more research, see [#open-issues](#open-issues)
+
+* use [`http.BasicAuth`](https://golang.org/pkg/net/http/#Request.BasicAuth) to handle the basic auth case
```

Comment:
> We should make sure that if adding auth affects dev routes we should change the APB tool or allow the 
> dev routes to bypass the auth.
>
How will this be implemented? I don't want us to create a code split on a config setting.  In other words, I think setting the dev route to bypass auth should be the dev turning off auth in the broker config.